core/market/decentralized/src/rest_api/provider.rs

```diff
@@ -43,7 +43,10 @@ async fn subscribe(
 
 #[actix_web::get("/offers")]
 async fn get_offers(market: Data<Arc<MarketService>>, id: Identity) -> HttpResponse {
-    HttpResponse::NotImplemented().finish()
+    match market.get_offers(Some(id)).await {
+        Ok(offers) => HttpResponse::Ok().json(offers),
+        Err(error) => HttpResponse::InternalServerError().json(&format!("{}", error)),
+    }
```

Comment:
Error should conform with API spec, which means it should return `ErrorMessage` serialized as JSON.
It is already handled by `OfferErorr` via implementing `ResponseError`. See `rest_api/error.rs`
```suggestion
    match market.get_offers(Some(id))
        .await
        .map(|offers| HttpResponse::Ok().json(offers))
    }
```